### PR TITLE
makeRelativeToProject

### PR DESCRIPTION
**Pull request by phadej**

### CODE REVIEW

File: th-compat.cabal [hunk 1]
```diff
@@ -46,6 +48,9 @@ library
   if !impl(ghc >= 8.0)
     build-depends:     fail             == 4.9.*
                      , transformers     >= 0.2 && < 0.7
+  if !impl(ghc >= 9.4)
+    build-depends:     filepath
```
Comment (phadej) [inline th-compat.cabal]: I forgot to add bounds here, will do later.

File: th-compat.cabal [hunk 2]
```diff
@@ -5,8 +5,9 @@ synopsis:            Backward- (and forward-)compatible Quote and Code types
 description:         This package defines a "Language.Haskell.TH.Syntax.Compat"
                      module, which backports the @Quote@ and @Code@ types to
                      work across a wide range of @template-haskell@ versions.
+                     @makeRelativeToProject@ utility is also backported.
```
Comment (RyanGlScott) [inline th-compat.cabal]:
```suggestion
                     The @makeRelativeToProject@ utility is also backported.
```

File: src/Language/Haskell/TH/Syntax/Compat.hs [hunk 3]
```diff
@@ -1059,3 +1072,56 @@ unTypeSplice = unTypeCode
...
+--
+-- This is best-effort compatibility implementation.
+-- This function looks at the source location of the Haskell file calling it,
+-- finds the first parent directory with a @.cabal@ file, and uses that as the
+-- root directory for fixing the relative path.
+--
+getPackageRoot :: Q FilePath
+getPackageRoot = getPackageRootPredicate $ (==) ".cabal" . takeExtension
+
+-- The implementation is modified version of one in file-embed package
```
Comment (RyanGlScott) [inline src/Language/Haskell/TH/Syntax/Compat.hs]:
```suggestion
-- The implementation is modified from the function of the same name in the file-embed package
```

Comment (phadej) [reply src/Language/Haskell/TH/Syntax/Compat.hs]: Err, it's not the same name there.

**Comment by phadej** [comment]
GitHub is really unhelpful, it wants you to approve every push for CI run :(

File: src/Language/Haskell/TH/Syntax/Compat.hs (hunk 3 above)
Comment (RyanGlScott) [reply src/Language/Haskell/TH/Syntax/Compat.hs]: Ah, right you are. Good catch!

**Review by RyanGlScott — APPROVED** [review]

**Pull request merged**